Packages/com.unity.inputsystem/InputSystem/Plugins/UI/MouseModel.cs

```diff
@@ -309,10 +303,36 @@ public void OnFrameFinished()
             m_MiddleButton.OnFrameFinished();
         }
 
+        public void CopyTo(PointerEventData eventData)
+        {
+            eventData.pointerId = pointerId;
+            eventData.position = position;
+            eventData.delta = deltaPosition;
+            eventData.scrollDelta = scrollDelta;
+
+            eventData.pointerEnter = m_InternalData.pointerTarget;
+            eventData.hovered.Clear();
+            eventData.hovered.AddRange(m_InternalData.hoverTargets);
+            
+            // This is unset in legacy systems and can safely assumed to stay true.
+            eventData.useDragThreshold = true;
+        }
+
+        public void CopyFrom(PointerEventData eventData)
+        {
+            InlinedArray<GameObject> hoverTargets = m_InternalData.hoverTargets;
+            hoverTargets.ClearWithCapacity();
+            hoverTargets.Append(eventData.hovered);
+            m_InternalData.hoverTargets = hoverTargets;
+            m_InternalData.pointerTarget = eventData.pointerEnter;
+        }
+
         private Vector2 m_Position;
         private Vector2 m_ScrollPosition;
         private MouseButtonModel m_LeftButton;
         private MouseButtonModel m_RightButton;
         private MouseButtonModel m_MiddleButton;
+
+        private InternalData m_InternalData;
```

Comment:
What's the advantage of keeping that in a struct rather than just making it member fields?

Reply:
Organizational.
If I expect others to use this, I think it's easier to get people to use it if I split 'Things you set' and 'Things uGUI sets' so that it's nor overwhelming.  Compiler side, it's the same, since it's a struct in a struct, but for users new to it, it increases legibility.